src/frogger.cpp

```diff
@@ -82,59 +125,68 @@ Rcpp::List run_base_model(const Rcpp::List data,
   // 0-based indexing vs R 1-based
   const int time_art_start =
       Rcpp::as<int>(projection_parameters["t_ART_start"]) - 1;
-  const leapfrog::TensorMap1<int> age_groups_hiv_span =
-      get_age_groups_hiv_span(projection_parameters, hiv_age_stratification);
-  int age_groups_hiv = static_cast<int>(age_groups_hiv_span.size());
+  const int age_groups_hiv = get_age_groups_hiv(hiv_age_stratification);
+
   int age_groups_hiv_15plus = age_groups_hiv;
   const int scale_cd4_mortality =
       Rcpp::as<int>(projection_parameters["scale_cd4_mort"]);
   int hIDX_15PLUS = 0;
   const double art_alloc_mxweight = Rcpp::as<double>(projection_parameters["art_alloc_mxweight"]);
 
-  const leapfrog::TensorMap2<double> base_pop(REAL(data["basepop"]), age_groups_pop,
-                                              num_genders);
+  std::function<std::string(const std::string&)> build_prop_name = [hiv_age_stratification](const std::string& property) {
```

Comment:
why is this a lambda? more generally, why do we need to do string manipulation here? I would generally be suggesting moving all of this sort of thing into the R layer, as this is likely to only get more painful

Reply:
Writing this out in some detail for my understanding too.

In the full model there are 66 ages in the HIV model. Using Jeff's variable names, these start at age `pIDX_HIVADULT` (= `15`) for `hAG` (= `66`) ages. Meaning the HIV part of the model runs from age 15 up to the 80+ open ended age group.

There are 2 ways to run the model, with coarse age stratified or full age stratified HIV population. If fully stratified input data is given for all the individual 66 HIV age groups e.g. age = 15, age = 16, age = 17, .... If we run coarse stratified input data is supplied for a range of age groups. With the current input data this is age ranges 15-16, 17-19, 20-24, 25-29, ... . 

The coarse age stratified model is used for ? and the full model for ? (will double check with Jeff)

At the moment here and in leapfrog all the input data is passed in, so we pass in e.g. `art_mort_full` and `art_mort_coarse` (both the coarse and full stratified input data) and an arg `hiv_age_stratification` which controls whether we use `art_mort_full` or `art_mort_coarse` etc. The names of all these inputs follow a common pattern `prop_full` or `prop_coarse` hence the string manipulation to pick the correct data based on the value of `hiv_age_stratification`.

I think you're right we can move this into R as it would keep the C++ wrapper a bit cleaner which is always nice! Will just double check with Jeff when we want to use the different types of fit